xbmc/addons/settings/AddonSettings.cpp

```diff
@@ -435,39 +437,11 @@ bool CAddonSettings::ParseSettingVersion(const CXBMCTinyXML& doc, uint32_t& vers
   return true;
 }
 
-bool CAddonSettings::InitializeFromOldSettingDefinitions(const CXBMCTinyXML& doc)
+std::shared_ptr<CSettingGroup> CAddonSettings::ParseOldSettingElement(const TiXmlElement* categoryElement, std::shared_ptr<CSettingCategory> category, std::set<std::string>& actionSettings)
 {
-  CLog::Log(LOGDEBUG, "CAddonSettings[%s]: trying to load setting definitions from old format...", m_addonId.c_str());
-
-  const TiXmlElement* root = doc.RootElement();
-  if (root == nullptr)
-    return false;
-
-  std::shared_ptr<CSettingSection> section = std::make_shared<CSettingSection>(m_addonId, GetSettingsManager());
-
-  std::shared_ptr<CSettingCategory> category;
-  const TiXmlElement *categoryElement = root->FirstChildElement("category");
-  // add a default category if necessary
-  if (categoryElement == nullptr)
-    categoryElement = root;
-
-  std::set<std::string> settingIds;
-  std::set<std::string> actionSettings;
-
-  uint32_t categoryId = 0;
-  while (categoryElement != nullptr)
-  {
-    // create the category
-    category = std::make_shared<CSettingCategory>(StringUtils::Format("category%u", categoryId), GetSettingsManager());
-    categoryId += 1;
-
-    // try to get the category's label and fall back to "General"
-    int categoryLabel = 128;
-    ParseOldLabel(categoryElement, g_localizeStrings.Get(categoryLabel), categoryLabel);
-    category->SetLabel(categoryLabel);
-
     // build a vector of settings from the same category
     std::vector<std::shared_ptr<const CSetting>> categorySettings;
+    std::set<std::string> settingIds;
```

Comment:
That means moving it outside the method where it's used. Too ugly for an unknown gain, likely not a hot path so I doubt the extra allocations would be noticable